web/src/stream_settings_ui.ts

```diff
@@ -76,6 +76,16 @@ export function update_left_panel_row(sub: StreamSubscription): void {
         $new_row.addClass("active");
     }
 
+    // Show spinner in $new_row if $row contains spinner.
+    const show_spinner = $($row).find(".sub_unsub_status").children().length > 0;
```

Comment:
It would be better to hide the spinner in `success` and `error` callbacks instead of here. Any reason for doing it here?